module/VuFind/src/VuFind/Controller/DeveloperSettingsController.php

```diff
@@ -0,0 +1,126 @@
+use VuFind\Exception\Forbidden;
+
+/**
+ * Controller for developer settings i.e API keys
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class DeveloperSettingsController extends AbstractBase
+{
+    /**
+     * Display developer settings
+     *
+     * @return mixed
+     */
+    public function displaySettingsAction()
+    {
+        if (!$user = $this->getUser()) {
+            return $this->forceLogin();
+        }
+        $developerSettingsService = $this->getService(DeveloperSettingsService::class);
+        if (!$developerSettingsService->apiKeysEnabled()) {
+            throw new Forbidden('Developer settings disabled.');
+        }
+        $view = $this->createViewModel();
+        $view->apiKeys = $developerSettingsService->getApiKeysForUser($user);
+        $view->createAllowed = !$developerSettingsService->apiKeysBlocked($view->apiKeys);
+        return $view;
+    }
+
+    /**
+     * Generate an API key for a user.
+     *
+     * @return mixed
+     */
+    public function generateApiKeyAction()
+    {
+        if (!$user = $this->getUser()) {
+            return $this->forceLogin();
+        }
+        $developerSettingsService = $this->getService(DeveloperSettingsService::class);
+        if (!$developerSettingsService->apiKeysEnabled() || !$this->permission()->isAuthorized('feature.Developer')) {
+            throw new Forbidden('Access denied.');
+        }
+
+        $view = $this->createViewModel();
+        if ($this->formWasSubmitted()) {
+            if ($title = $this->fromPostAndQuery('title')) {
+                if ($apiKey = $developerSettingsService->generateApiKeyForUser($user, $title)) {
+                    $successMsg = $this->translate(
+                        'Developer::api_key_generation_success',
+                        ['%%TOKEN%%' => $apiKey->getToken()]
+                    );
+                    $this->flashMessenger()->addSuccessMessage($successMsg);
+                    return $view;
+                }
+            }
+            $this->flashMessenger()->addErrorMessage('An error has occurred');
+        }
+
+        return $view;
+    }
+
+    /**
+     * Delete an API key for a user.
+     *
+     * @return mixed
+     */
+    public function deleteApiKeyAction()
+    {
+        if (!$user = $this->getUser()) {
+            return $this->forceLogin();
+        }
+        $developerSettingsService = $this->getService(DeveloperSettingsService::class);
+        if (!$developerSettingsService->apiKeysEnabled() || !$this->permission()->isAuthorized('feature.Developer')) {
+            throw new Forbidden('Access denied.');
+        }
+        if ($this->fromPostAndQuery('confirm') === '1') {
+            $id = $this->fromPostAndQuery('id');
+            if (
+                false !== $id
```

Comment:
Since fromPostAndQuery returns null by default rather than false, I think this needs to be adjusted:

```suggestion
                null !== $id
```